planner_cspace/src/planner_3d.cpp

```diff
@@ -1796,17 +1830,25 @@ class Planner3dNode
 
   void diagnoseStatus(diagnostic_updater::DiagnosticStatusWrapper& stat)
   {
-    if (status_.error == planner_cspace_msgs::PlannerStatus::IN_ROCK)
-    {
-      stat.summary(diagnostic_msgs::DiagnosticStatus::ERROR, "The robot is in rock.");
-    }
-    else if (status_.error == planner_cspace_msgs::PlannerStatus::PATH_NOT_FOUND)
+    switch (status_.error)
     {
-      stat.summary(diagnostic_msgs::DiagnosticStatus::ERROR, "Path not found.");
-    }
-    else
-    {
-      stat.summary(diagnostic_msgs::DiagnosticStatus::OK, "Going well.");
+      case planner_cspace_msgs::PlannerStatus::GOING_WELL:
+        break;
```

Comment:
@at-wat 
No message shown in /diagnostics if PlannerStatus::GOING_WELL?

- going well
```
$ rostopic echo /diagnostics

header: 
  seq: 107
  stamp: 
    secs: 1543971395
    nsecs: 467201380
  frame_id: ''
status: 
  - 
    level: 2
    name: "planner_3d: Path Planner Status"
    message: "No message was set"
    hardware_id: "none"
    values: 
      - 
        key: "status"
        value: "1"
      - 
        key: "error"
        value: "0"
---

```

- cf. path not found
```
---
header: 
  seq: 278
  stamp: 
    secs: 1543971505
    nsecs: 666996955
  frame_id: ''
status: 
  - 
    level: 2
    name: "planner_3d: Path Planner Status"
    message: "Path not found."
    hardware_id: "none"
    values: 
      - 
        key: "status"
        value: "1"
      - 
        key: "error"
        value: "2"
---
```

Reply:
I have added a check for DATA_MISSING and GOING_WELL diagnostics status in the test.
Test for the other cases of diagnostics outputs might better be added.